### PR TITLE
NDI-41: Fixed report page layout

### PR DESCRIPTION
**Pull request by mukeshcompucorp**
# Description
This PR
- Styles report template instances to make it look better
- Adds padding to footer to make it look unified on all pages of civicrm
- Adds new mixin `input-fields-container`

# Screenshots
**Before:**
image
**After:**
image

**Before:**
image
**After:**
image

**Before:**
image
**After:**
image

# Technical Details
New mixin `input-fields-container` is introduces to handle all selectors for input field containers at one place,
```scss
@mixin input-fields-container {
  .form-layout,
  .form-layout-compressed,
  .crm-accordion-body,
  .crm-form-block,
  #task-section {
    @content;
  }
}
```
# Comments
- BackstopJS tested
- Browserstack tested
- Stylelint tested

### CODE REVIEW

File: scss/civicrm/common/_forms.scss [hunk 1]
```diff
@@ -109,7 +109,8 @@
   .form-layout,
   .form-layout-compressed,
   .crm-accordion-body,
-  .crm-form-block {
+  .crm-form-block,
+  #task-section {
```
Comment (AkA84) [inline scss/civicrm/common/_forms.scss]: what's going on with this list? the elements in this list seems to be very different from each (forms, accordions, sections), what is it that they all have in common?

Comment (mukeshcompucorp) [reply scss/civicrm/common/_forms.scss]: This are the selectors under which inputs can be there in CiviCRM,
Basically we are targeting each selectors which has input fields, and in CiviCRM it's not always that the inputs are wrapped with `form` tag

Comment (mukeshcompucorp) [reply scss/civicrm/common/_forms.scss]: Also, we can not directly style on `form` tag as, we are injecting with Resource URL so, no matter what, page it is, Admin or Public, our css will reflect and, it can have potential break on front parts

Comment (AkA84) [reply scss/civicrm/common/_forms.scss]: ok could you then write a mixin that you then document? something like
```scss
@include form-containers { // find a more descriptive name
  // the style
}
```

and then mixin will be something like
```scss
@mixin form-containers {
  .form-layout,
  .form-layout-compressed,
  .crm-accordion-body,
  .crm-form-block,
  #task-section {
    @content;
  }
}

File: scss/civicrm/mixins/_input-fields-container.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/* stylelint-disable max-nesting-depth, selector-max-compound-selectors, selector-no-qualifying-type, selector-max-id */
+
+@mixin input-fields-container {
```
Comment (AkA84) [inline scss/civicrm/mixins/_input-fields-container.scss]: you need to document the mixin, that was the whole point

File: scss/civicrm/common/_forms.scss [hunk 3]
```diff
@@ -480,6 +379,88 @@
     }
   }
 
+  %responsive {
```
Comment (AkA84) [inline scss/civicrm/common/_forms.scss]: why do we have these placeholders lying around? don't we have a folder for them?

File: scss/civicrm/common/_forms.scss [hunk 4]
```diff
@@ -701,6 +682,22 @@
       color: $crm-white;
     }
   }
+
+  //Page specific
```
Comment (AkA84) [inline scss/civicrm/common/_forms.scss]: shouldn't we have a partial just for this pages then?

Comment (mukeshcompucorp) [reply scss/civicrm/common/_forms.scss]: No, we don't have any directory for placeholders and there are many placeholders scattered around different styles.

File: scss/civicrm/common/_forms.scss (hunk 3 above)
Comment (mukeshcompucorp) [reply scss/civicrm/common/_forms.scss]: No, we don't have any directory for placeholders and there are many placeholders scattered around different styles.

File: scss/civicrm/common/_forms.scss (hunk 4 above)
Comment (mukeshcompucorp) [reply scss/civicrm/common/_forms.scss]: Yes, that makes sense, I never had look on here. Caught this due to linter changing orders

**Comment by jamienovick** [comment]
@mukeshcompucorp where CiviCRM is inconsistent could we fix the inconsistencies too with a PR to core?

**Comment by mukeshcompucorp** [comment]
@jamienovick yes we are doing that as well, see my PRs [here](https://github.com/civicrm/civicrm-core/pulls?utf8=%E2%9C%93&q=is%3Apr+author%3Amukeshcompucorp).
Though, Core CiviCRM is already working fine, so, we only fix the thing which seems potentially wrong on shoreditch and civicrm for now.

File: scss/civicrm/common/_forms.scss [hunk 5]
```diff
@@ -701,6 +680,7 @@
       color: $crm-white;
     }
   }
+
```
Comment (AkA84) [inline scss/civicrm/common/_forms.scss]: unnecessary line

File: scss/civicrm/contact/_contact.scss [hunk 6]
```diff
@@ -1,7 +1,13 @@
+/* stylelint-disable max-nesting-depth, selector-max-compound-selectors, selector-no-qualifying-type, selector-max-id */
+
 #{civi-page('contact-view')} {
   @import 'tabs';
   @import 'header';
   @import 'detail';
+
+  .select2-search::before {
```
Comment (AkA84) [inline scss/civicrm/contact/_contact.scss]: why is this needed only for this page?

Comment (mukeshcompucorp) [reply scss/civicrm/contact/_contact.scss]: This page has `margin-top: 8px` on `input[type='text']` so, it needs counter css for fixing search icon position and make it look proper.
https://github.com/civicrm/org.civicrm.shoreditch/blob/master/scss/civicrm/contact/_detail.scss#L779-L784

Screenshot:
image

Comment (AkA84) [reply scss/civicrm/contact/_contact.scss]: > This page has `margin-top: 8px` on `input[type='text']`

why?

Comment (mukeshcompucorp) [reply scss/civicrm/contact/_contact.scss]: All popups which open in Contact view popups are styled that way.
There are many popups such as below having such style. It is 2 year old styling.
image

I just moved this style from common/_forms.scss to this page specific now. In near future, when I work on popups, this will be removed and no longer needed.

File: scss/civicrm/common/_forms.scss (hunk 5 above)
Comment (mukeshcompucorp) [reply scss/civicrm/common/_forms.scss]: Fixed this now
